### PR TITLE
feat(sdks): Add sdkMetadata to Clerk singleton, populate it for each SDK

### PR DESCRIPTION
**Pull request by BRKalow**
## Description

Adds a new static property to the Clerk singleton, `sdkMetadata`, and populates it for each host SDK. We'll use this data to make it easier to understand what host SDK is initializing `Clerk`.

```
> Clerk.sdkMetadata
{name: '@clerk/nextjs', version: '4.25.3'}
```

As part of adding this to each SDK, I also introduced `tsup` to the remaining packages that weren't already using it. This allows us to define the `PACKAGE_NAME` and `PACKAGE_VERSION` constants at build time:
* `remix`
* `gatsby-plugin-clerk`
* `expo`

Note that there's definitely an opportunity to consolidate some of our package tooling, but I'm deferring that work for now.

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [x] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [x] `@clerk/clerk-react`
- [x] `@clerk/nextjs`
- [x] `@clerk/remix`
- [x] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [x] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [x] `@clerk/fastify`
- [x] `@clerk/chrome-extension`
- [x] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: packages/react/src/isomorphicClerk.ts [hunk 1]
```diff
@@ -198,6 +198,8 @@ export default class IsomorphicClerk {
         await global.Clerk.load(this.options);
       }
 
+      global.Clerk.sdkMetadata = this.options.sdkMetadata ?? { name: PACKAGE_NAME, version: PACKAGE_VERSION };
```
Comment (BRKalow) [inline packages/react/src/isomorphicClerk.ts:201]: For SDKs that use `ClerkProvider`, this is how we are passing through and setting the metadata. SDKs that access `Clerk` directly assign it themselves.

File: packages/gatsby-plugin-clerk/src/GatsbyClerkProvider.tsx [hunk 2]
```diff
@@ -7,6 +7,13 @@ import {
 import { navigate } from 'gatsby';
 import React from 'react';
 
+import { name, version } from '../package.json';
+
+const SDK_METADATA = {
+  name,
+  version,
+};
```
Comment (BRKalow) [inline packages/gatsby-plugin-clerk/src/GatsbyClerkProvider.tsx:15]: We are importing from `package.json` directly here as this package isn't built via `tsup`, so we can't define the global variable at build-time.

File: packages/expo/package.json [hunk 3]
```diff
@@ -21,8 +21,11 @@
     "dist"
   ],
   "scripts": {
-    "build": "tsc -p tsconfig.build.json",
-    "dev": "tsc -p tsconfig.build.json --watch",
+    "build": "tsup",
+    "dev": "tsup --watch",
+    "dev:publish": "npm run dev -- --env.publish",
+    "build:declarations": "tsc -p tsconfig.declarations.json",
```
Comment (LekoArts) [inline packages/expo/package.json:27]: I know that we do this in the other packages but IMO we should just use `tsup` to also output the types and not use TSC

But that can happen in another PR

File: packages/expo/tsconfig.json [hunk 4]
```diff
@@ -1,6 +1,26 @@
 {
-  "extends": "./tsconfig.build.json",
   "compilerOptions": {
+    "outDir": "dist",
```
Comment (LekoArts) [inline packages/expo/tsconfig.json:3]: We should extend this from a root/base tsconfig.json

But that's for another PR

**Review by LekoArts — APPROVED** [review]

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**